bindgen/src/formatter.ts

```diff
@@ -22,17 +22,42 @@ import cp from "child_process";
 import { extend } from "./debug";
 const debug = extend("format");
 
-const FORMATTERS: Record<string, string[]> = {
-  eslint: ["npx", "eslint", "--fix", "--format=stylish"],
-  "clang-format": ["npx", "clang-format", "-i"],
+type Formatter = (cwd: string, filePaths: string[]) => void;
+
+export function executeCommand(cwd: string, command: string, ...args: string[]) {
+  console.log(chalk.dim(command, ...args));
+  const result = cp.spawnSync(command, args, {
+    cwd,
+    encoding: "utf8",
+    stdio: "inherit",
+    shell: true,
+  });
+  if (result.error) throw result.error;
+  if (result.status) {
+    throw new Error(`Exited with status ${result.status}`)
+  }
+}
+
+function createCommandFormatter(commandAndArgs: string[]): Formatter {
+  return (cwd, filePaths) => {
+    const [command, ...args] = [...commandAndArgs, ...filePaths];
+    executeCommand(cwd, command, ...args);
```

Comment:
Isn't this the same as `executeCommand(cwd, ...commandAndArgs, ...filePaths);`?

Reply:
From a type perspective both `commandAndArgs` and `filePaths` could be empty, in which case `executeCommand` would be called with `undefined` as the second argument, which is not allowed.

Reply:
So is this just a roundabout way of asserting that commandAndArgs isn't empty? Could we assert that more directly, or is that not enough to make tsc happy?

Alternatively, consider making `executeCommand()` take `args` as a single array rather than as varargs. It seems silly to expand the array just to repack it into another array.

Reply:
This changed in my latest commit 👍